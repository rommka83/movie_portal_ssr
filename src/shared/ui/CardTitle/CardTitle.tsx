import React, { HTMLAttributes } from 'react';
import styles from './cardtitle.module.css';
import classNames from 'classnames';

export const CardTitle = ({ className, children, ...props }: CardTitleProps) => {
  return (
    <h3 {...props} className={classNames(styles.cardTitle, className)}>
      {children}
    </h3>
  );
};

type CardTitleProps = HTMLAttributes<HTMLHeadingElement> & {
  className?: string;
};
