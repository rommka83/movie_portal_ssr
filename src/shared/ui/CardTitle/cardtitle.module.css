.cardTitle {
  font-size: 15px;
  line-height: 20px;
  font-weight: 600;
  color: var(--gray-7e0);
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  cursor: pointer;
}
