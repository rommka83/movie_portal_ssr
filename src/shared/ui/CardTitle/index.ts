export { CardTitle as default } from "./CardTitle";
