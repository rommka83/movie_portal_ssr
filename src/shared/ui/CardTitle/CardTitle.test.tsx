import renderer from 'react-test-renderer';
import { CardTitle } from './CardTitle';

describe('проверка CardTitle', () => {
  test('отрисовка CardTitle', () => {
    const component = renderer.create(
      <CardTitle>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Tenetur quod velit rem consequuntur nihil
        ullam, dicta modi nostrum dolore saepe illum provident quae asperiores excepturi possimus, fuga,
        distinctio officia numquam?
      </CardTitle>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
