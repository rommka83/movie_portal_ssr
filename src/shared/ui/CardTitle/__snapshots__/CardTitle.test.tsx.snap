// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка CardTitle отрисовка CardTitle 1`] = `
<h3
  className="cardTitle"
>
  Lorem ipsum dolor, sit amet consectetur adipisicing elit. Tenetur quod velit rem consequuntur nihil ullam, dicta modi nostrum dolore saepe illum provident quae asperiores excepturi possimus, fuga, distinctio officia numquam?
</h3>
`;
